src/connmgr.cpp

```diff
@@ -2,17 +2,179 @@
 // Copyright (c) 2009-2015 The Bitcoin Core developers
 // Copyright (c) 2016-2017 The Bitcoin Unlimited developers
 
-#include <atomic>
-
 #include "connmgr.h"
+#include "expedited.h"
 
+std::unique_ptr<CConnMgr> connmgr(new CConnMgr);
 
-NodeId CConnMgr::nextNodeId()
+/**
+ * Find a node in a vector.  Just calls std::find.  Split out for cleanliness and also because std:find won't be
+ * enough if we switch to vectors of noderefs.
+ */
+static std::vector<CNode *>::iterator FindNode(std::vector<CNode *> &vNodes, CNode *pNode)
 {
-    static std::atomic<NodeId> next;
+    return std::find(vNodes.begin(), vNodes.end(), pNode);
+}
 
+/**
+ * Add a node to a vector, and take a reference.  This function does NOT prevent adding duplicates.
+ * @param[in] vNodes      The vector of nodes.
+ * @param[in] pNode       The node to add.
+ */
+static void AddNode(std::vector<CNode *> &vNodes, CNode *pNode)
+{
+    pNode->AddRef();
+    vNodes.push_back(pNode);
```

Comment:
They're utility methods that traverse the vector; it should be obvious that any lock needed for vector traversal is necessary, but I guess it can't hurt to say it explicitly.  I'll add that.